mne/time_frequency/tfr.py

```diff
@@ -2092,15 +2092,50 @@ def __abs__(self):
         epochs.data = np.abs(self.data)
         return epochs
 
-    def average(self):
+    def average(self, method='mean'):
         """Average the data across epochs.
 
+        Parameters
+        ----------
+        method : str | callable
+            How to combine the data. If "mean"/"median", the mean/median
+            are returned. Otherwise, must be a callable which, when passed
+            an array of shape (n_epochs, n_channels, n_freqs, n_time)
+            returns an array of shape (n_channels, n_freqs, n_time).
+            Note that due to file type limitations, the kind for all
+            these will be "average".
+
         Returns
         -------
         ave : instance of AverageTFR
             The averaged data.
+
+        Notes
+        -----
+        Passing in ``median`` is considered unsafe when there is complex
+        data. Use with caution. We use the marginal median in the
+        complex case.
```

Comment:
can you clarify this? should we even accept it then? What is marginal median, can you better explain 
or provide a ref?

Reply:
Passing in the string `'median'` should be fine, internally we should notice this and compute the marginal median. Passing in `np.median` is unsafe (and should be documented as unsafe here) because NumPy don't compute the marginal median, they sort the complex values by real part and return whatever value is computed.

The marginal median is the median of each component separately, which is easy to compute: 

https://en.wikipedia.org/wiki/Median#Marginal_median

This is the way we are going for SciPy:

https://github.com/scipy/scipy/pull/12676#issuecomment-783370228

Reply:
Updated docstring in 9ad5741